src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Interfaces/IReleaseDataFileService.cs

```diff
@@ -31,15 +31,21 @@ public interface IReleaseDataFileService
             Guid releaseVersionId,
             List<Guid> fileIds);
 
-        Task<Either<ActionResult, DataFileInfo>> Upload(Guid releaseVersionId,
+        Task<Either<ActionResult, DataFileInfo>> Upload(
+            Guid releaseVersionId,
             IFormFile dataFormFile,
             IFormFile metaFormFile,
-            Guid? replacingFileId = null,
-            string? subjectName = null);
+            string? dataSetTitle = null,
```

Comment:
`Upload` and `UploadAsZip` must be provided either a `dataSetTitle` or a `replacingFIleId`. If it's an original import, it uses the `dataSetTitle`. If it's a replacement, it uses the `dataSetTitle` of `ReleaseFile` being replaced.

`GetReleaseVersionDataSetTitle` ensures that either one or the other is provided in the arguments.

If your query is specifically "Why make them optional arguments?", I think that's just because that's what the code was originally. I think both are always supplied (in non-test code, at least) when the methods are called. 

I've assumed removing the default value was the change you had in mind so I've removed the default value for both args.